source/extensions/filters/http/cache/cache_filter.h

```diff
@@ -99,6 +106,10 @@ class CacheFilter : public Http::PassThroughFilter,
   // https://httpwg.org/specs/rfc7234.html#response.cacheability
   bool request_allows_inserts_ = false;
 
+  // These are used to keep track of whether we should fetch more data from the cache.
+  int high_watermark_calls_ = 0;
+  bool ongoing_fetch_ = false;
```

Comment:
Optional: ongoing_fetch_ and high_watermark_calls_ could be combined into one counter with a name like pause_counter_. Saves a few bytes and removes a condition in maybeGetBody.

Reply:
I think it's more semantically clear to have them like this.